### ISSUE TITLE
No late binding with __future__ and Python 3.7

### ISSUE THREAD

**Issue by rosbly** [microsoft/pyright#585]
**Describe the bug**
With Python 3.7 and using `from __future__ import annotations`, when refering from a dataclass definition, to types declared later in the source file, pyright reports a binding error.

**Expected behavior**
pyright should allow late binding when using "annotations" future feature, in line with Python 3.7

**To reproduce**
```python
from __future__ import annotations
from dataclasses import dataclass

@dataclass
class ClassA:
    hello: ClassB

class ClassB:
    pass

b = ClassB()
a = ClassA(b)
```

![annotations](https://user-images.githubusercontent.com/40209070/77535539-0882d680-6e9b-11ea-8114-8b65f56c502e.png)

**VS Code extension or command-line**
Using pyright 1.1.29 as a VS Code extension, VSCode 1.43.2.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next published version of Pyright.

**Comment by erictraut** [comment]
This is implemented in 1.1.30

**Comment by rosbly** [comment]
Thank you!
